Library/Homebrew/download_strategy.rb

```diff
@@ -483,7 +483,13 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
 
       if (filename_with_encoding = content_disposition.parameters["filename*"])
         encoding, encoded_filename = filename_with_encoding.split("''", 2)
-        filename = URI.decode_www_form_component(encoded_filename).encode(encoding) if encoding && encoded_filename
+        # If the `filename*` has incorrectly added double quotes, e.g.
+        #   content-disposition: attachment; filename="myapp-1.2.3.pkg"; filename*=UTF-8''"myapp-1.2.3.pkg"
+        # Then the encoded_filename will come back as the empty string, in which case we should fall back to the
+        # `filename` parameter.
+        if encoding && encoded_filename.present?
```

Comment:
```suggestion
        if encoding.present? && encoded_filename.present?
```

Reply:
@MikeMcQuaid I am noticing an issue here with https://github.com/Homebrew/homebrew-cask-drivers/pull/3360
If the conditional changes to `if encoding.present? && encoded_filename` or `if encoding.present? && encoded_filename.blank?` it works as expected.
With `.present?` I'm seeing an error `Error: no implicit conversion of nil into String` thrown at Line `497`.

CC: @gibfahn

Reply:
The current logic is correct but we should probably adjust for the case where we can't parse the filename properly, e.g.:

```rb
filename ||= content_disposition.filename
next if filename.nil?

File.basename(filename)
```

Reply:
@gibfahn @bevanjkay @Bo98 could one of you open a PR for this? Thanks!